### PR TITLE
Enhance id comparision for updateAttributes

### PR DESCRIPTION
**Pull request by raymondfeng**
/to @bajtos or @ritch 

string/number and MongoDB ObjectID equality are now allowed.

### CODE REVIEW

File: lib/dao.js [hunk 1]
```diff
@@ -1908,8 +1929,9 @@ DataAccessObject.prototype.updateAttributes = function updateAttributes(data, op
   var idNames = Model.definition.idNames();
   for (var i = 0, n = idNames.length; i < n; i++) {
     var idName = idNames[i];
-    if (data[idName] !== undefined && data[idName] !== inst[idName]) {
-      var err = new Error('id property value cannot be updated: ' + idName);
+    if (data[idName] !== undefined && !idEquals(data[idName], inst[idName])) {
+      var err = new Error('id property (' + idName +
+        ') cannot be updated from ' + inst[idName] + ' to ' + data[idName]);
```
Comment (bajtos) [inline lib/dao.js]: Nitpick: The formatting is a bit strange, seeing a closing brace at the start of a line feels weird:

File: test/manipulation.test.js [hunk 2]
```diff
@@ -503,6 +503,25 @@ describe('manipulation', function () {
...
+    it('should allow same stringified id value on updateAttributes',
+      function(done) {
+        var pid = person.id;
+        if (typeof person.id === 'object' || typeof person.id === 'number') {
+          // For example MongoDB ObjectId
+          pid = person.id.toString();
+        }
+        person.updateAttributes({id: pid, name: 'John'},
+          function(err, p) {
+            should.not.exist(err);
```
Comment (bajtos) [inline test/manipulation.test.js]: Please use `if (err) return done(err)`, it gives better failure messages in the test output. Same comment applies to L517.

**Comment by bajtos** [comment]
Two minor comments, the rest LGTM.

**Comment by raymondfeng** [comment]
Issues addressed.

**Pull request merged**